### DISCUSSION TITLE
Why when trying to run tauri for android it fails because OpenSsl was not found?

### DISCUSSION THREAD

**Discussion by Bicheka** [tauri-apps/tauri#11020] [Q&A] [answered]
I have a tauri app with that i want to run on mobile so I am moving to v2 and I am trying to run npx tauri android dev but i get this error
`
cargo:rerun-if-env-changed=X86_64_LINUX_ANDROID_OPENSSL_LIB_DIR
  X86_64_LINUX_ANDROID_OPENSSL_LIB_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_LIB_DIR
  OPENSSL_LIB_DIR unset
  cargo:rerun-if-env-changed=X86_64_LINUX_ANDROID_OPENSSL_INCLUDE_DIR
  X86_64_LINUX_ANDROID_OPENSSL_INCLUDE_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_INCLUDE_DIR
  OPENSSL_INCLUDE_DIR unset
  cargo:rerun-if-env-changed=X86_64_LINUX_ANDROID_OPENSSL_DIR
  X86_64_LINUX_ANDROID_OPENSSL_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_DIR
  OPENSSL_DIR unset
  cargo:rerun-if-env-changed=OPENSSL_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64-linux-android
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_x86_64_linux_android
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-linux-android
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_linux_android
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-linux-android
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_linux_android
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  run pkg_config fail: pkg-config has not been configured to support cross-compilation.

  Install a sysroot for the target platform and configure it via
  PKG_CONFIG_SYSROOT_DIR and PKG_CONFIG_PATH, or install a
  cross-compiling wrapper for pkg-config and set it via
  PKG_CONFIG environment variable.

  --- stderr
  thread 'main' panicked at C:\Users\biche\.cargo\registry\src\index.crates.io-6f17d22bba15001f\openssl-sys-0.9.103\build\find_normal.rs:190:5:

  Could not find directory of OpenSSL installation, and this `-sys` crate cannot

  Make sure you also have the development packages of openssl installed.
  For example, `libssl-dev` on Ubuntu or `openssl-devel` on Fedora.

  If you're in a situation where you think the directory *should* be found
  automatically, please open a bug at https://github.com/sfackler/rust-openssl
  and include information about your system as well as this message.

  $HOST = x86_64-pc-windows-msvc
  $TARGET = x86_64-linux-android
  openssl-sys = 0.9.103
`
I already have android studio and every requirement of that the documentation says but this just does not work. I am using windows 11 how can I fix this to be able run tauri for android.

**Comment by FabianLars** [comment]
Are you using Rust crates that don't come with the default templates and require TLS/SSL? In my opinion the best way forward is to identify those dependencies and change them from openssl to rustls. Most crates should have feature flags for that.

If this doesn't work for you, you could try adding
```toml
openssl-sys = { version = "0.9", featuers = ["vendored"] }
```
and see if that's enough. iirc this requires `perl` and `make` on your system though (which is a problem if you can only build on Windows...)

**Reply by Bicheka** [reply]
I did my research and I was using reqwest crate, I changed the dependency to use TLS `[dependencies]
reqwest = { version = "0.11", default-features = false, features = ["rustls-tls"] }` and the app compiled but instead of showing up the ui it shows an error message in the app screen
![Screenshot 2024-09-16 151001](https://github.com/user-attachments/assets/35c85193-9f76-46f6-a6c2-207b98ade7b0)

**Reply by FabianLars** [reply, marked as answer]
This should be a frontend misconfiguration. Today i had the same error. Not sure exactly what should happen / what we did wrong here, but if you're using vite, open the vite config file and replace `host: ...` with `host: '0.0.0.0'` and try again.

**Reply by Bicheka** [reply]
It works, thanks man I really appreciate your help and for answering so quickly
